src/client/game/errors/errGeneric.ts

```diff
@@ -0,0 +1,25 @@
+import { GameError } from "./error";
+
+export default class ErrGeneric extends Error implements GameError {
```

Comment:
Some new errors are defined in this PR; most of these are unrecoverable (meaning the user will be prompted to try and re-join the game/refresh the page). `ErrGeneric` is the most generic with these, allowing the caller to specify whether it is recoverable or not (no by default).